app/domain/authentication/authn_jwt/signing_key/public_signing_keys.rb

```diff
@@ -0,0 +1,42 @@
+
+module Authentication
+  module AuthnJwt
+    module SigningKey
+      # This class is a POJO class presents public-keys structure
+      class PublicSigningKeys
+        include ActiveModel::Validations, AttrRequired
+
+        VALID_TYPES =  %w(jwks).freeze
+
+        attr_required(:type, :value)
+
+        validates *required_attributes, presence: true
+        validates :type, inclusion: { in: VALID_TYPES, message: "'%{value}' is not a valid public-keys type" }
+        validate :validate_value_is_jwks, if: -> { @type == "jwks" }
+
+        def initialize(hash)
+          raise Errors::Authentication::AuthnJwt::InvalidPublicKeys.new("the value is not in valid JSON format") unless
+            hash.is_a?(Hash)
+          hash = hash.with_indifferent_access
+          required_attributes.each do |key|
+            self.send "#{key}=", hash[key]
+          end
+        end
+
+        def validate!
```

Comment:
Why the ! in the function name? It is a ruby convention?

Reply:
Methods that are "dangerous," or modify the receiver, might be named with a trailing `!`. ([ref](http://ruby-doc.com/docs/ProgrammingRuby/html/tut_methods.html))
The method throws exception in case and the object is in invalid state.
I've considered [this](https://github.com/nov/openid_connect/blob/4acd4802a79217c0124dde43f9233df678616a6b/lib/openid_connect/discovery/provider/config/response.rb#L80) example from openid library.